packages/core/src/Services/socket-general.js

```diff
@@ -142,8 +143,13 @@ const BinarySocketGeneral = (() => {
             const remaining_session_time = duration * 60 * 1000 - current_session_duration;
             clearTimeout(session_timeout);
             session_timeout = setTimeout(() => {
-                client_store.logout();
-                sessionStorage.removeItem('session_start_time');
+                OAuth2Logout({
+                    WSLogoutAndRedirect: () => {
+                        client_store.logout();
+                        sessionStorage.removeItem('session_start_time');
+                    },
+                    redirectCallbackUri: `${window.location.origin}/callback`,
```

Comment:
This works even if growthbook is turned off, so we don't need the Growthbook check here

Reply:
In `useOauth2` we don't have Growthbook checks for logout as well
<img width="690" alt="Screenshot 2025-02-26 at 12 56 13 PM" src="https://github.com/user-attachments/assets/2bd509f2-edf2-4d7f-9bfc-5c7a8e7b350e" />